ls1intum/Artemis
# Add hazelcast configuration options.

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test servers 2 and 4

### Motivation and Context
Artemis should be able to be deployed in an IPv6-environment. Therefore, Hazelcast should support binding to specific interfaces.

### Description
Added options to configure the Hazelcast port (defaults to `5701`), the interface to bind to (defaults to all) and whether to connect to local instances (on the same host) or remote instances on a different host (defaults to remote).

### Steps for Testing
Set the following values in the config:
`eureka.instance.ip-address` to the IP-address that should be populated to the JHipster registry (e.g. `[2a02:810d:4ac0:2778:b54b:c900:9247:a15e]`).

Set the value of `spring.hazelcast.interface` e.g. to `2a02:810d:4ac0:2778:b54b:c900:9247:a15e` (without the brackets).
Set the value of `spring.hazelcast.localInstances`to `true` in case you are running the setup locally.

1. Verifiy that once two members find each other, the log statement looks like this

 ```
Members {size:2, ver:2} [
	Member [2a02:810d:4ac0:2778:b54b:c900:9247:a15e]:13781 - 0b70d62e-7cb8-4614-8974-b24f8f8c6471 this
	Member [2a02:810d:4ac0:2778:b54b:c900:9247:a15e]:13782 - 28e12c88-8261-4ca1-8a5a-3cfc51cc58d6
]
```
, so verify that IPv6 addresses are used.


This patch has already been tested on the test server 4, where several VMs are in place.
I've also verified that is still works with local machines both on my machine and on the test server 2.
